url8af84d296f_mrcosta_gantt_generator_tgz-pJ8-org_gantt_generator_mrcpsp_RunJ8/src/org/swiftgantt/common/CalendarUtils.java

```diff
@@ -0,0 +1,293 @@
+package org.swiftgantt.common;
+
+import org.apache.commons.lang.time.DateUtils;
+import org.apache.log4j.LogManager;
+import org.apache.log4j.Logger;
+
+import org.swiftgantt.Config;
+import java.util.GregorianCalendar;
+
+/**
+ * Common utils for Time.
+ * 
+ * @author Yuxing Wang
+ * @version 1.0
+ * 
+ */
+public class CalendarUtils {
+	//这里所有的方法都可以重构到Time类里面。
+
+	static Logger logger = null;
+
+
+	static {
+		logger = LogManager.getLogger(CalendarUtils.class);
+	}
+	static final int[] DAY_OF_MONTH = new int[12];
+
+
+	static {
+		DAY_OF_MONTH[0] = 31;
+		DAY_OF_MONTH[1] = 28;// 29 when leap year.
+		DAY_OF_MONTH[2] = 31;
+		DAY_OF_MONTH[3] = 30;
+		DAY_OF_MONTH[4] = 31;
+		DAY_OF_MONTH[5] = 30;
+		DAY_OF_MONTH[6] = 31;
+		DAY_OF_MONTH[7] = 31;
+		DAY_OF_MONTH[8] = 30;
+		DAY_OF_MONTH[9] = 31;
+		DAY_OF_MONTH[10] = 30;
+		DAY_OF_MONTH[11] = 31;
+	}
+
+	/**
+	 * Get amount of days in a month
+	 * 
+	 * @param month
+	 * @return int
+	 * @deprecated to Time
+	 */
+	public static int getDaysOfMonth(Time Time) {
+		if (Time == null) {
+			throw new NullPointerException("No time provided");
+		}
+		int year = Time.get(Time.YEAR);
+		int month = Time.get(Time.MONTH);
+		if (month == 1 && ((GregorianCalendar) Time).isLeapYear(year)) {
+			return 29;
+		}
+		return DAY_OF_MONTH[month];
+	}
+
+	/**
+	 * Truncate time by given Time filed, exclude any Week.
+	 * @param time
+	 * @param field The field of Time to keep.
+	 * @deprecated to Time
+	 * @return
+	 */
+	public static void truncateTime(Time time, int field) {
+		switch (field) {
+			case Time.YEAR:
+				time.set(Time.MONTH, 0);
+			case Time.MONTH:
+				time.set(Time.DAY_OF_MONTH, 1);
+//		case Time.WEEK_OF_YEAR:
+//			time.set(Time.DAY_OF_WEEK, Time.SUNDAY);
+			case Time.DAY_OF_MONTH:
+				time.set(Time.HOUR_OF_DAY, 0);
+			case Time.HOUR:
+			case Time.HOUR_OF_DAY:
+				time.set(Time.MINUTE, 0);
+		}
+		time.set(Time.SECOND, 0);
+		time.set(Time.MILLISECOND, 0);
+	}
+
+	/**
+	 * 
+	 * @param start
+	 * @param dates
+	 * @return
+	 */
+	public static int calcWorkingDays(Time start, int dates) {
+		Time end = start.clone().increaseDates(dates);
+//		end.add(Time.DATE, dates);
+		return calcWorkingDays(start, end, null);
+	}
+
+	/**
+	 * Calculate working days from start day to end day(incluing end day). workingDurationInWeek represent the index of start working day
+	 * and end working day in a week.
+	 * 
+	 * @param start
+	 * @param end
+	 * @param workingDurationInWeek
+	 * @return
+	 */
+	public static int calcWorkingDays(Time start, Time end, int[] workingDurationInWeek) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 14. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 141-159 into a separate method.